### PR TITLE
feat(timelinenav)!: remove top-level sub-component

### PR DESCRIPTION
**Pull request by jinlee93**
### Summary:
- merge TimelineNavPanel into TimelineNav directly

### Test Plan:

- [x] CI tests / new tests are not applicable
  - No snap changes but also LP specific so low risk

### CODE REVIEW

File: src/components/TimelineNav/TimelineNav.tsx [hunk 1]
```diff
@@ -51,18 +49,79 @@ export interface Props {
...
-export interface TimelineNavItem {
-  props: {
-    'aria-label': string;
-    children: ReactNode;
-    completed?: boolean;
-    title?: string;
-    titleAfter?: ReactNode;
-    variant?: TimelineNavPanelVariant;
-  };
-}
```
Comment (jinlee93) [inline src/components/TimelineNav/TimelineNav.tsx:65]: Never actually used 🤷

File: src/components/TimelineNav/TimelineNav.tsx [hunk 2]
```diff
@@ -51,18 +49,79 @@ export interface Props {
...
+type TimelineNavPanelVariant =
+  | 'bullet'
+  | 'complete'
+  | 'error'
+  | 'incomplete'
+  | 'number'
+  | 'success'
+  | 'warning'
+  | undefined;
```
Comment (booc0mtaco) [inline src/components/TimelineNav/TimelineNav.tsx:62]: This is a little funny; if a prop is set to `undefined` react will ignore it. Adding undefined to a type is a bit ambiguous. Instead it's better to use `?` for marking that.

I'm guessing this is inherited tho, so not requesting a change.

**Review by booc0mtaco — APPROVED** [review]
LGTM

File: src/components/TimelineNav/TimelineNav.tsx (hunk 2 above)
Comment (jinlee93) [reply src/components/TimelineNav/TimelineNav.tsx:62]: Will change to `?` as using `undefined` is dumb 👍

**Pull request merged**